src/containers/WrongChainRecovery/WrongChainRecovery.tsx

```diff
@@ -1,4 +1,141 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { CoinsSelectAutocomplete } from '~/components/CoinsSelectAutocomplete';
+import { useAlertBanner } from '~/contexts';
+import { mapSdkErrorToAlert, safeEnv } from '~/helpers';
+import {
+  allCoinMetas,
+  CoinMetadata,
+  wrongChainRecoveryCoins,
+} from '~/helpers/config';
+import { WrongChainRecoveryForm } from './WrongChainRecoveryForm';
+
 export function WrongChainRecovery() {
-  // TODO(BG-67209)
-  return <>{'Coming Soon'}</>;
+  const { env } = useParams<'env'>();
+  const navigate = useNavigate();
+  const [_, setAlert] = useAlertBanner();
+
+  const environment = safeEnv(env);
+
+  const sourceCoins: readonly CoinMetadata[] = Object.keys(
+    wrongChainRecoveryCoins[environment]
+  ).map(key => {
+    return allCoinMetas[key as keyof typeof allCoinMetas];
+  });
+  const [sourceCoin, setSourceCoin] = useState<string>(sourceCoins[0].value);
+
+  const destinationCoins = useMemo(() => {
+    return wrongChainRecoveryCoins[environment][sourceCoin];
+  }, [environment, sourceCoin]);
+  const [destinationCoin, setDestinationCoin] = useState<string>(
+    destinationCoins[0].value
+  );
+
+  useEffect(() => {
+    if (
+      destinationCoins.length &&
+      !destinationCoins.find(item => item.value === destinationCoin)
+    ) {
+      setDestinationCoin(destinationCoins[0].value);
+    }
+  }, [destinationCoins, destinationCoin]);
```

Comment:
just confused as to the use case for this effect? I'm guessing it's to reset the `destinationCoin` if destinationCoin selected isn't the appropriate?

Reply:
its to set destinationCoin to valid value whenever the list of destinationCoins changes. 

I.e. 
selectedCoin: BTC
destinationCoins: [BCH, LTC]
destinationCoin LTC

If you then change selectedCoin to BCHA,
destinationCoins will change to [BCH]
LTC is not valid value, this effect will reset the form back to BCH as a default value.

Reply:
can't help but think there's a better way to do this without having the guard on `useEffect` but nothing's coming to mind.

Reply:
The alternative is the onChange prop for the source coin, but this just makes that CB more complicated then it needs to be.